src/pool/pool_jemalloc.c

```diff
@@ -72,7 +74,14 @@ static void *arena_extent_alloc(extent_hooks_t *extent_hooks, void *new_addr,
         return NULL;
     }
 
+#ifndef __SANITIZE_ADDRESS__
+    // jemalloc might write to new extents in realloc, so we cannot
+    // mark them as unaccessible under asan
+    utils_annotate_memory_unaccessible(ptr, size);
```

Comment:
why not undefined? I think this is not consistent with the memset() called later where you mark mem as defined